launcher/config_test.go

```diff
@@ -72,8 +78,58 @@ func (logger *testLogger) requireNotContains(t *testing.T, expected string) {
 	}
 }
 
-func (logger *testLogger) reset() {
-	logger.output = nil
+// Create some dummy server implementations so that we can
+// spin up tests that don't need to wait for a timeout trying to send data.
+type dummyTraceServer struct {
+	collectortrace.UnimplementedTraceServiceServer
+}
+
+func (*dummyTraceServer) Export(ctx context.Context, req *collectortrace.ExportTraceServiceRequest) (*collectortrace.ExportTraceServiceResponse, error) {
+	fmt.Println("inside trace export", time.Now().Format(time.RFC3339))
+	return &collectortrace.ExportTraceServiceResponse{}, nil
+}
+
+type dummyMetricsServer struct {
+	collectormetrics.UnimplementedMetricsServiceServer
+}
+
+func (*dummyMetricsServer) Export(ctx context.Context, req *collectormetrics.ExportMetricsServiceRequest) (*collectormetrics.ExportMetricsServiceResponse, error) {
+	fmt.Println("inside metrics export", time.Now().Format(time.RFC3339))
+	return &collectormetrics.ExportMetricsServiceResponse{}, nil
+}
+
+// dummyGRPCListener is a test helper that builds a dummy grpc server that does nothing but
+// returns quickly so that we don't have to wait for timeouts
+func dummyGRPCListener() func() {
+	grpcServer := grpc.NewServer()
+	collectortrace.RegisterTraceServiceServer(grpcServer, &dummyTraceServer{})
+	collectormetrics.RegisterMetricsServiceServer(grpcServer, &dummyMetricsServer{})
+
+	// we need to listen on localhost, not 0.0.0.0, to satisfy JAMF without causing problems
+	l, err := net.Listen("tcp", net.JoinHostPort("localhost", "4317"))
+	if err != nil {
+		panic("oops - dummyGrpcListener failed to start up!")
+	}
+	go func() {
+		_ = grpcServer.Serve(l)
+	}()
+	return grpcServer.Stop
+}
+
+// withTestExporters conforms to the Option interface and sets up the options needed
+// to prevent a test from having to time out. It won't work unless the test also does this:
+//
+// stopper := dummyGRPCListener()
+// defer stopper()
+//
+// This is a convenience function.
+func withTestExporters() Option {
+	return func(c *Config) {
+		WithSpanExporterEndpoint("localhost:4317")(c)
```

Comment:
The launcher should default to export to `localhost:4317`. It's fine to have them explicit here, but just making the note.